### ISSUE TITLE
Does function caffe_rng_gaussian generate truncated random value in a range [min_value,max_value]?

### ISSUE THREAD

**Issue by dianyancao** [BVLC/caffe#4219]
I look at the code in "caffe/util/math_functions.cpp" and search for boost::normal_distribution and  boost::variate_generator.
But there I can't confirm function caffe_rng_gaussian does truncate the generate random value in a proper range?

**Comment by seanbell** [comment]
Why would you expect it to truncate it?  It's a Gaussian variable, which is unbounded.

**Comment by ajtulloch** [comment]
It does not. Feel free to send a PR for truncated normal initialization though.

**Comment by dianyancao** [comment]
@seanbell @ajtulloch So when use normal_distribution to initialize the weights in network,then there maybe some very large value in the weights?

**Comment by seanbell** [comment]
Sure, there could be some large values with low probability.  I don't think this should ever be an issue though, especially given that weight decay would shrink those weights very quickly.

**Comment by ajtulloch** [comment]
If you find it's a problem, consider adding the equivalent of https://www.tensorflow.org/versions/r0.8/api_docs/python/constant_op.html#truncated_normal, which would be a trivial few line change to math_functions.cpp if you're interested.

**Comment by dianyancao** [comment]
Yeah,use weight decay may solve it,and I will also change it to truncated values.
Thank you.
